Selenium.Axe/Selenium.Axe/AxeBuilder.cs

```diff
@@ -95,47 +154,36 @@ public AxeResult Analyze(IWebElement context)
         /// <returns>An aXe results document</returns>
         public AxeResult Analyze()
         {
-            string axeContext;
+            bool runContextHasData = runContext.Include?.Any() == true || runContext.Exclude?.Any() == true;
 
-            if (_includeExcludeManager.HasMoreThanOneSelectorsToIncludeOrSomeToExclude())
-            {
-                axeContext = $"{ _includeExcludeManager.ToJson()}";
-                //command =    $"axe.a11yCheck({ _includeExcludeManager.ToJson()}, {Options}, arguments[arguments.length - 1]);";
-            }
-            else if (_includeExcludeManager.HasOneItemToInclude())
+            string contextToBeSent = runContextHasData ? JsonConvert.SerializeObject(runContext, JsonSerializerSettings) : null;
```

Comment:
Should we keep the "default to document" behavior here, rather than defaulting to null and making `scan.js` responsible for that logic? It's easier to test if it's here...

Reply:
i cannot pass it as a string. i would have to somehow get the document object from webdriver & pass it.

---
In reply to: [311818483](https://github.com/devaradhanm/SeleniumAxeDotnet/pull/1#discussion_r311818483) [](ancestors = 311818483)